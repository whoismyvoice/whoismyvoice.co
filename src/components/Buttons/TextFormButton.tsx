import * as React from 'react';
import TextButton from './TextButton';

class TextFormButton extends TextButton {
  render(): JSX.Element {
    const { text } = this.props;
    const { buttonClasses } = this.state;
    return (
      <button
        type="submit"
        className={buttonClasses}
        onClick={this.props.onClick}
      >
        {text}
        <span className="text-button-border" />
      </button>
    );
  }
}

export default TextFormButton;
