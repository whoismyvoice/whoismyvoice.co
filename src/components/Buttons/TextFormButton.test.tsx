import * as React from 'react';
import * as ReactDOM from 'react-dom';
import * as renderer from 'react-test-renderer';
import TextFormButton from './TextFormButton';

jest.mock('mixpanel-browser');

const props = {
  text: 'text',
};

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TextFormButton {...props} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const tree = renderer.create(<TextFormButton {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
