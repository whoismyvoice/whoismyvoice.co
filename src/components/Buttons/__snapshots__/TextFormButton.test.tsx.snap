// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="text-button"
  onClick={[Function]}
  type="submit"
>
  text
  <span
    className="text-button-border"
  />
</button>
`;
